File: packages/frontend/src/types/xiansuo.ts
/**
 * 线索管理相关类型定义
 */

// 线索来源
export interface XiansuoLaiyuan {
  id: string
  laiyuan_mingcheng: string
  laiyuan_bianma: string
  laiyuan_leixing: string
  huoqu_chengben: number
  xiansuo_shuliang: number
  zhuanhua_shuliang: number
  zhuanhua_lv: number
  zhuangtai: string
  paixu: number
  miaoshu?: string
  created_at: string
  updated_at: string
}

export interface XiansuoLaiyuanCreate {
  laiyuan_mingcheng: string
  laiyuan_bianma: string
  laiyuan_leixing: string
  huoqu_chengben?: number
  zhuangtai?: string
  paixu?: number
  miaoshu?: string
}

export interface XiansuoLaiyuanUpdate {
  laiyuan_mingcheng?: string
  laiyuan_bianma?: string
  laiyuan_leixing?: string
  huoqu_chengben?: number
  zhuangtai?: string
  paixu?: number
  miaoshu?: string
}

// 线索状态
export interface XiansuoZhuangtai {
  id: string
  zhuangtai_mingcheng: string
  zhuangtai_bianma: string
  zhuangtai_leixing: string
  shangyige_zhuangtai?: string
  xiayige_zhuangtai?: string
  yanse_bianma: string
  tubiao_mingcheng?: string
  shi_zhongzhong_zhuangtai: string
  shi_chenggong_zhuangtai: string
  paixu: number
  zhuangtai: string
  miaoshu?: string
  created_at: string
  updated_at: string
}

export interface XiansuoZhuangtaiCreate {
  zhuangtai_mingcheng: string
  zhuangtai_bianma: string
  zhuangtai_leixing: string
  shangyige_zhuangtai?: string
  xiayige_zhuangtai?: string
  yanse_bianma?: string
  tubiao_mingcheng?: string
  shi_zhongzhong_zhuangtai?: string
  shi_chenggong_zhuangtai?: string
  paixu?: number
  zhuangtai?: string
  miaoshu?: string
}

// 线索跟进记录
export interface XiansuoGenjin {
  id: string
  xiansuo_id: string
  genjin_fangshi: string
  genjin_shijian: string
  genjin_neirong: string
  kehu_fankui?: string
  kehu_taidu?: string
  xiaci_genjin_shijian?: string
  xiaci_genjin_neirong?: string
  genjin_jieguo?: string
  genjin_ren_id: string
  genjin_ren_xingming?: string
  fujian_lujing?: string
  created_at: string
  updated_at: string
}

export interface XiansuoGenjinCreate {
  xiansuo_id: string
  genjin_fangshi: string
  genjin_shijian?: string
  genjin_neirong: string
  kehu_fankui?: string
  kehu_taidu?: string
  xiaci_genjin_shijian?: string
  xiaci_genjin_neirong?: string
  genjin_jieguo?: string
  fujian_lujing?: string
}

export interface XiansuoGenjinUpdate {
  genjin_fangshi?: string
  genjin_shijian?: string
  genjin_neirong?: string
  kehu_fankui?: string
  kehu_taidu?: string
  xiaci_genjin_shijian?: string
  xiaci_genjin_neirong?: string
  genjin_jieguo?: string
  fujian_lujing?: string
}

// 线索主表
export interface Xiansuo {
  id: string
  xiansuo_bianma: string
  gongsi_mingcheng: string
  lianxi_ren: string
  lianxi_dianhua?: string
  lianxi_youxiang?: string
  hangye_leixing?: string
  gongsi_guimo?: string
  zhuce_dizhi?: string
  fuwu_leixing?: string
  yusuan_fanwei?: string
  shijian_yaoqiu?: string
  xiangxi_xuqiu?: string
  zhiliang_pinggu: string
  zhiliang_fenshu: number
  laiyuan_id: string
  laiyuan_xiangxi?: string
  xiansuo_zhuangtai: string
  fenpei_ren_id?: string
  fenpei_shijian?: string
  shouci_genjin_shijian?: string
  zuijin_genjin_shijian?: string
  xiaci_genjin_shijian?: string
  genjin_cishu: number
  shi_zhuanhua: string
  zhuanhua_shijian?: string
  zhuanhua_jine: number
  kehu_id?: string
  created_at: string
  updated_at: string
}

export interface XiansuoCreate {
  gongsi_mingcheng: string
  lianxi_ren: string
  lianxi_dianhua?: string
  lianxi_youxiang?: string
  hangye_leixing?: string
  gongsi_guimo?: string
  zhuce_dizhi?: string
  fuwu_leixing?: string
  yusuan_fanwei?: string
  shijian_yaoqiu?: string
  xiangxi_xuqiu?: string
  zhiliang_pinggu?: string
  zhiliang_fenshu?: number
  laiyuan_id: string
  laiyuan_xiangxi?: string
}

export interface XiansuoUpdate {
  gongsi_mingcheng?: string
  lianxi_ren?: string
  lianxi_dianhua?: string
  lianxi_youxiang?: string
  hangye_leixing?: string
  gongsi_guimo?: string
  zhuce_dizhi?: string
  fuwu_leixing?: string
  yusuan_fanwei?: string
  shijian_yaoqiu?: string
  xiangxi_xuqiu?: string
  zhiliang_pinggu?: string
  zhiliang_fenshu?: number
  laiyuan_id?: string
  laiyuan_xiangxi?: string
}

export interface XiansuoDetail extends Xiansuo {
  laiyuan?: XiansuoLaiyuan
  genjin_jilu_list: XiansuoGenjin[]
}

// 线索状态更新
export interface XiansuoStatusUpdate {
  xiansuo_zhuangtai: string
  beizhu?: string
}

// 线索分配更新
export interface XiansuoAssignUpdate {
  fenpei_ren_id: string
  beizhu?: string
}

// 线索统计
export interface XiansuoStatistics {
  total_xiansuo: number
  new_xiansuo: number
  following_xiansuo: number
  interested_xiansuo: number
  quoted_xiansuo: number
  won_xiansuo: number
  lost_xiansuo: number
  zhuanhua_lv: number
  pingjun_zhuanhua_zhouzqi: number
  pingjun_zhuanhua_jine: number
}

// 线索报价相关类型
export interface XiansuoBaojiaXiangmu {
  id: string
  baojia_id: string
  chanpin_xiangmu_id: string
  xiangmu_mingcheng: string
  shuliang: number
  danjia: number
  danwei: string
  xiaoji: number
  paixu: number
  beizhu?: string
  created_at: string
  updated_at: string
}

export interface XiansuoBaojia {
  id: string
  xiansuo_id: string
  baojia_bianma: string
  baojia_mingcheng: string
  zongji_jine: number
  baojia_zhuangtai: string
  youxiao_qi: string
  beizhu?: string
  is_expired: boolean
  xiangmu_list: XiansuoBaojiaXiangmu[]
  created_at: string
  updated_at: string
  created_by: string
}

export interface XiansuoBaojiaXiangmuInput {
  chanpin_xiangmu_id: string
  xiangmu_mingcheng?: string
  shuliang: number
  danjia?: number
  danwei?: string
  paixu?: number
  beizhu?: string
}

export interface XiansuoBaojiaCreate {
  xiansuo_id: string
  baojia_mingcheng: string
  youxiao_qi: string
  beizhu?: string
  xiangmu_list: XiansuoBaojiaXiangmuInput[]
}

export interface XiansuoBaojiaUpdate {
  baojia_mingcheng?: string
  youxiao_qi?: string
  baojia_zhuangtai?: string
  beizhu?: string
  xiangmu_list?: XiansuoBaojiaXiangmuInput[]
}

export interface XiansuoBaojiaListItem {
  id: string
  baojia_bianma: string
  baojia_mingcheng: string
  zongji_jine: number
  baojia_zhuangtai: string
  youxiao_qi: string
  is_expired: boolean
  xiangmu_count: number
  created_at: string
  created_by: string
}

export interface XiansuoInfoForBaojia {
  id: string
  gongsi_mingcheng: string
  lianxi_ren: string
  lianxi_dianhua?: string
  lianxi_youxiang?: string
  kehu_id?: string
}

export interface XiansuoBaojiaDetail {
  id: string
  xiansuo_id: string
  baojia_bianma: string
  baojia_mingcheng: string
  youxiao_qi: string
  zongji_jine: number
  baojia_zhuangtai: string
  is_expired: boolean
  beizhu?: string
  xiangmu_list: XiansuoBaojiaXiangmu[]
  xiansuo_info: XiansuoInfoForBaojia
  created_at: string
  updated_at: string
  created_by: string
}

export interface XiansuoBaojiaListParams {
  page?: number
  size?: number
  xiansuo_id?: string
  baojia_zhuangtai?: string
  search?: string
}

export interface XiansuoBaojiaListResponse {
  items: XiansuoBaojiaListItem[]
  total: number
  page: number
  size: number
  pages: number
}

export interface ChanpinFenleiOption {
  id: string
  fenlei_mingcheng: string
  chanpin_leixing: string
}

export interface ChanpinXiangmuOption {
  id: string
  xiangmu_mingcheng: string
  xiangmu_bianma: string
  yewu_baojia: number
  baojia_danwei: string
  banshi_tianshu: number
  fenlei_id: string
}

export interface ChanpinDataForBaojia {
  zengzhi_fenlei: ChanpinFenleiOption[]
  daili_jizhang_fenlei: ChanpinFenleiOption[]
  zengzhi_xiangmu: ChanpinXiangmuOption[]
  daili_jizhang_xiangmu: ChanpinXiangmuOption[]
}

// API 响应类型
export interface XiansuoListResponse {
  items: Xiansuo[]
  total: number
  page: number
  size: number
}

// 旧版兼容类型（如有遗留调用可逐步迁移）
export interface LegacyXiansuoBaojiaListResponse {
  items: XiansuoBaojia[]
  total: number
  page: number
  size: number
}

export interface XiansuoLaiyuanListResponse {
  items: XiansuoLaiyuan[]
  total: number
  page: number
  size: number
}

export interface XiansuoZhuangtaiListResponse {
  items: XiansuoZhuangtai[]
  total: number
  page: number
  size: number
}

export interface XiansuoGenjinListResponse {
  items: XiansuoGenjin[]
  total: number
  page: number
  size: number
}

// 查询参数类型
export interface XiansuoListParams {
  page?: number
  size?: number
  search?: string
  xiansuo_zhuangtai?: string
  laiyuan_id?: string
  fenpei_ren_id?: string
  zhiliang_pinggu?: string
  hangye_leixing?: string
  start_date?: string
  end_date?: string
}

export interface XiansuoLaiyuanListParams {
  page?: number
  size?: number
  search?: string
  laiyuan_leixing?: string
  zhuangtai?: string
}

export interface XiansuoGenjinListParams {
  page?: number
  size?: number
  xiansuo_id?: string
  genjin_ren_id?: string
  genjin_fangshi?: string
  start_date?: string
  end_date?: string
}

File: packages/backend/src/models/xiansuo_guanli/xiansuo.py
"""
线索主表模型
"""
from sqlalchemy import Column, String, DateTime, Text, Integer, ForeignKey
from sqlalchemy.types import Numeric
from sqlalchemy.orm import relationship
from ..base import BaseModel

class Xiansuo(BaseModel):
    """线索表"""
    
    __tablename__ = "xiansuo"
    __table_args__ = {"comment": "线索表"}
    
    # 基本信息
    xiansuo_bianma = Column(
        String(50),
        unique=True,
        nullable=False,
        comment="线索编码"
    )
    
    gongsi_mingcheng = Column(
        String(200),
        nullable=False,
        comment="公司名称"
    )
    
    lianxi_ren = Column(
        String(50),
        nullable=False,
        comment="联系人"
    )
    
    lianxi_dianhua = Column(
        String(20),
        nullable=True,
        comment="联系电话"
    )
    
    lianxi_youxiang = Column(
        String(100),
        nullable=True,
        comment="联系邮箱"
    )
    
    # 公司信息
    hangye_leixing = Column(
        String(100),
        nullable=True,
        comment="行业类型"
    )
    
    gongsi_guimo = Column(
        String(50),
        nullable=True,
        comment="公司规模：small-小型，medium-中型，large-大型"
    )
    
    zhuce_dizhi = Column(
        String(500),
        nullable=True,
        comment="注册地址"
    )
    
    # 需求信息
    fuwu_leixing = Column(
        String(200),
        nullable=True,
        comment="服务类型"
    )
    
    yusuan_fanwei = Column(
        String(100),
        nullable=True,
        comment="预算范围"
    )
    
    shijian_yaoqiu = Column(
        String(200),
        nullable=True,
        comment="时间要求"
    )
    
    xiangxi_xuqiu = Column(
        Text,
        nullable=True,
        comment="详细需求"
    )
    
    # 线索质量评估
    zhiliang_pinggu = Column(
        String(20),
        default="medium",
        comment="质量评估：high-高质量，medium-中等质量，low-低质量"
    )
    
    zhiliang_fenshu = Column(
        Integer,
        default=0,
        comment="质量分数（0-100）"
    )
    
    # 来源信息
    laiyuan_id = Column(
        String(36),
        ForeignKey("xiansuo_laiyuan.id"),
        nullable=False,
        comment="来源ID"
    )
    
    laiyuan_xiangxi = Column(
        String(500),
        nullable=True,
        comment="来源详细信息"
    )
    
    # 状态信息
    xiansuo_zhuangtai = Column(
        String(50),
        default="new",
        nullable=False,
        comment="线索状态"
    )
    
    # 分配信息
    fenpei_ren_id = Column(
        String(36),
        nullable=True,
        comment="分配人ID（销售人员）"
    )
    
    fenpei_shijian = Column(
        DateTime,
        nullable=True,
        comment="分配时间"
    )
    
    # 跟进信息
    shouci_genjin_shijian = Column(
        DateTime,
        nullable=True,
        comment="首次跟进时间"
    )
    
    zuijin_genjin_shijian = Column(
        DateTime,
        nullable=True,
        comment="最近跟进时间"
    )
    
    xiaci_genjin_shijian = Column(
        DateTime,
        nullable=True,
        comment="下次跟进时间"
    )
    
    genjin_cishu = Column(
        Integer,
        default=0,
        comment="跟进次数"
    )
    
    # 转化信息
    shi_zhuanhua = Column(
        String(1),
        default="N",
        comment="是否转化 Y/N"
    )
    
    zhuanhua_shijian = Column(
        DateTime,
        nullable=True,
        comment="转化时间"
    )
    
    zhuanhua_jine = Column(
        Numeric(10, 2),
        default=0.00,
        comment="转化金额"
    )
    
    kehu_id = Column(
        String(36),
        nullable=True,
        comment="转化后的客户ID"
    )
    
    # 关联关系
    laiyuan = relationship(
        "XiansuoLaiyuan",
        back_populates="xiansuo_list"
    )
    
    # 注意：状态关联通过编码字段关联，不是外键关联
    
    genjin_jilu_list = relationship(
        "XiansuoGenjin",
        back_populates="xiansuo",
        cascade="all, delete-orphan"
    )

    baojia_list = relationship(
        "XiansuoBaojia",
        back_populates="xiansuo",
        cascade="all, delete-orphan",
        order_by="XiansuoBaojia.created_at.desc()"
    )
